src/analysis/complex/circle.lean

```diff
@@ -97,3 +99,76 @@ def exp_map_circle_hom : ℝ →+ (additive circle) :=
   map_zero' := by { rw exp_map_circle, convert of_mul_one, simp },
   map_add' := λ x y, show exp_map_circle (x + y) = (exp_map_circle x) * (exp_map_circle y),
     from subtype.ext $ by simp [exp_map_circle, exp_add, add_mul] }
+
+lemma exp_map_circle_eq_one_iff {x : ℝ} :
+  exp_map_circle x = 1 ↔ x ∈ add_subgroup.gmultiples (2 * π) :=
+begin
+  simp [add_subgroup.mem_gmultiples_iff],
+  split,
+  { simp[exp_map_circle],
+    intro h,
+    apply_fun (coe: circle → ℂ) at h,
+    simp at h,
+    rw exp_eq_one_iff at h,
+    cases h with n hn,
+    rw mul_comm at hn,
+    rw mul_comm (↑n) (2 * ↑π * I) at hn,
+    rw mul_assoc at hn,
+    rw mul_assoc at hn,
+    rw mul_comm (↑π) (I * ↑n) at hn,
+    rw mul_comm (2) (I * ↑n * ↑π) at hn,
+    rw mul_assoc at hn,
+    rw mul_assoc at hn,
+    rw mul_right_inj' I_ne_zero at hn,
+    rw mul_comm (↑π) (2 : ℂ) at hn,
+    use n,
+    apply eq.symm,
+    exact_mod_cast hn, },
+  { intro h,
+    cases h with k hk,
+    rw ← hk,
+    ext1,
+    simp[exp_map_circle],
+    rw exp_eq_one_iff,
+    use k,
+    simp[mul_assoc], },
+end
+
+lemma exp_map_circle_surjective : function.surjective exp_map_circle :=
+begin
+  intros z,
+  have this₀ : (complex.log z).re = 0,
+  { rw complex.log_re,
+    rw abs_eq_of_mem_circle,
+    simp, },
+  use (log(z)).im,
+  ext1,
+  simp [exp_map_circle_hom],
+  have : ↑(complex.log z).im * I = log z,
+  { rw ← re_add_im (log z),
+    rw this₀,
+    simp, },
+  rw this,
+  exact complex.exp_log (nonzero_of_mem_circle z),
+end
+
+/-- The additive-group isomorphism identifying `real.angle` with the additive version of the
+`circle` group. -/
+def angle_to_circle_hom : real.angle ≃+ additive circle :=
+(quotient_add_group.equiv_quotient_of_eq (by { ext x, exact exp_map_circle_eq_one_iff.symm})).trans
+(quotient_add_group.quotient_ker_equiv_of_surjective exp_map_circle_hom exp_map_circle_surjective)
+
+/-- The equivalence identifying `real.angle` with the circle group. -/
+def angle_to_circle : real.angle ≃ circle := angle_to_circle_hom.to_equiv
```

Comment:
```suggestion
def angle_to_circle : real.angle ≃ circle := angle_to_circle_hom.to_equiv.trans additive.to_mul
```
otherwise if anyone does `rw [angle_to_circle]` they will get something nasty.